link/src/main/java/com/stripe/android/link/LinkActivityViewModel.kt

```diff
@@ -59,47 +70,88 @@ internal class LinkActivityViewModel @Inject internal constructor(
     ) : ViewModelProvider.Factory, Injectable<Factory.FallbackInitializeParam> {
         internal data class FallbackInitializeParam(
             val application: Application,
+            val starterArgs: LinkActivityContract.Args,
             val enableLogging: Boolean,
             val publishableKey: String,
             val stripeAccountId: String?,
             val productUsage: Set<String>
         )
 
         @Inject
-        lateinit var subComponentBuilderProvider:
-            Provider<LinkViewModelSubcomponent.Builder>
+        lateinit var viewModel: LinkActivityViewModel
+
+        private lateinit var injector: Injector
```

Comment:
Instead of buffering Injector, is it safe(that is we're sure those ViewModels are certainly instantiated after the Injector is available) to just pass the injectorKey to other `ViewModel` and let them call `requireNotNull(WeakMapInjectorRegistry.retrieve(key))` instead? This way at least we don't need to expose the actual `Injector` interface everywhere

Reply:
No, I tried that but there's no guarantee that the `WeakMapInjectorRegistry` will still be holding the `Injector`, since it's a weak reference. In this case the `Injector` needs to be held strongly somewhere, and `LinkActivityViewModel` is the best place since it's alive during the whole flow.